controllers/servers/host_definer/storage_manager/host_definer_server.py

```diff
@@ -44,4 +44,21 @@ def define_host(self, request):  # pylint: disable=invalid-name
             return DefineHostResponse(error_message=str(ex))
 
     def undefine_host(self, request):  # pylint: disable=invalid-name
-        raise NotImplementedError
+        node_id_info = NodeIdInfo(request.node_id)
+        initiators = node_id_info.initiators
+        try:
+            array_connection_info = get_array_connection_info_from_secrets(request.system_info)
+            array_type = detect_array_type(array_connection_info.array_addresses)
```

Comment:
can't we "cache" this in the k8s host definition resource?

Reply:
@matancarmeli7 what do you think?

Reply:
CSI-4603